vs_utils/utils/grid_factory.py

```diff
@@ -0,0 +1,148 @@
+from nnscore_pdb import PDB 
+from copy import deepcopy
+import numpy as np
+import pickle
+import csv
+import os
+from collections import defaultdict
+
+class GridGenerator:
+  def __init__(self):
+    self.box_x = 16.0
+    self.box_y = 16.0
+    self.box_z = 16.0
+    self.x_linspace = None
+    self.y_linspace = None
+    self.z_linspace = None
+    self.box = None
+    self.grid_res = 0.5
+    self.grid = {}
+    self.grid_vector = []
+    self.num_features = 3
+
+    path = os.path.dirname(os.path.realpath(__file__))
+
+    with open(os.path.join(path, 'info/periodic_table.p'), 'rb') as handle:
+      self.periodic_table = pickle.load(handle)
+    self.periodic_table = dict((k.upper(),v) for k,v in self.periodic_table.iteritems())
+
+    with open(os.path.join(path, 'info/protein_heavy_atoms.p'), 'rb') as handle:
+      self.protein_heavy_atoms = pickle.load(handle)
+
+    with open(os.path.join(path, 'info/all_atom_names.p'), 'rb') as handle:
+      self.all_atom_names = pickle.load(handle)
+    self.terminal_atoms = ["H1", "1H", "HN1", "HN1","HT1","HNCAP","H2", "2H", "HN2","HN2","HT2","H3","3H","HN3","HN3","HT3",
+                "OT1", "OXT", "OT2"]
+    self.all_atom_names += self.terminal_atoms
+    self.all_atom_names = list(set(self.all_atom_names))
+
+
+  def transform(self, box, box_x, box_y, box_z, voxel_width, grid_pickle_file, num_features=3):
+    '''
+    Takes as input a pickle file of a "box" generated from PDBTransformer's transform() function. This "box" will be an
+    instance of class PDB defined in nnscore_pdb.py. It will contain all atoms within a rectangular prism with edge lengths 
+    defined as box_x, box_y, and box_z. These dimensions must be the same as the dimensions fed to PDBTransformer.
+    [To do: maybe make "box" a class that inherits PDB?]
+
+    You must also provide: voxel_width, a float that describes in angstroms the edge length of a single voxel.
+    grid_pickle_file: pickle file for saving 4d np array of featurized grid
+    num_features: this will specify the length of the 4th dimension of the tensor. Default is 3.
+
+    '''
+
+    self.box_x = float(box_x)
```

Comment:
This is kind of messy. ```transform``` is acting as a de-factor constructor by setting a number of instance variables. Can you move this setting behavior up into the constructor?